src/Test/WindowsFormsIntegration/FeatureTests/WinFormsTestLib/Util/RandomUtil.cs

```diff
@@ -0,0 +1,3465 @@
+        // </param>
+        // <retvalue>
+        //  A random value of type System.Int64 between the specified minimum
+        //  and maximum values.
+        // </retvalue>
+        // </doc>
+        public virtual long GetInt64(long min, long max)
+        {
+            if (max < min)
+                throw new ArgumentOutOfRangeException("max must be greater than min");
+
+            // There is no way to method on System.Random to get a long so we
+            // need to calculate one ourselves.
+            //
+            // Stuff everything into a Decimal to maintain precision.
+            //
+            Int64 retVal = (long)((decimal)GetDouble() * ((decimal)max - (decimal)min) + (decimal)min);
+            LogValueIfLogging(retVal);
+            return retVal;
+        }
+
+        // <doc>
+        // <desc>
+        //  Retrieves a random boolean value
+        // </desc>
+        // <retvalue>
+        //  Randomly true or false
+        // </retvalue>
+        // </doc>
+        public virtual bool GetBoolean()
+        {
+            bool retVal = (this.GetInt() % 2 == 0);
+            LogValueIfLogging(retVal);
+            return retVal;
+        }
+
+        // <doc>
+        // <desc>
+        //  Retrieves a random DateTime value.
+        // </desc>
+        // <retvalue>
+        //  A random value of type System.DateTime.
+        // </retvalue>
+        // </doc>
+        public virtual DateTime GetDateTime()
+        {
+            DateTime retVal = GetDateTime(DateTime.MinValue, DateTime.MaxValue);
+            LogValueIfLogging(retVal);
+            return retVal;
+        }
+
+        // <doc>
+        // <desc>
+        //  Retrieves a random DateTime value within a given range.
+        // </desc>
+        // <param term="min">
+        //  The minimuim value.
+        // </param>
+        // <param term="max">
+        //  The maximum value.
+        // </param>
+        // <retvalue>
+        //  A random value of type System.DateTime between the specified minimum
+        //  and maximum values.
+        // </retvalue>
+        // </doc>
+        public virtual DateTime GetDateTime(DateTime min, DateTime max)
+        {
+            if (min > max)
+                throw new ArgumentOutOfRangeException("max must be greater than min");
+
+            DateTime retVal = new DateTime(GetInt64(min.Ticks, max.Ticks));
+            LogValueIfLogging(retVal);
+            return retVal;
+        }
+
+        //
+        // In some locales which use different calendars (e.g. Arabic, which uses the Hijiri calendar),
+        // the range of valid DateTimes is smaller than the standard Gregorian calendar.  This method
+        // will return a DateTime that should be valid for the current culture.
+        //
+        // HACK: In v1, there is no direct way to query a calendar for its min and max dates.  So we
+        //       determine the min date by creating a DateTime for 1/1/0001 in the current culture's
```

Comment:
@anjali-wpf can we confirm if this hack is to be mentioned and if the date is required.